src/styles/main.css

```diff
@@ -0,0 +1,199 @@
+body {
+  display: flex;
+  align-items: center;
+  justify-content: center;
+  min-width: 750px;
+}
+
+.calendar {
+  box-sizing: border-box;
+  display: flex;
+  flex-wrap: wrap;
+  width: calc(7 * 100px + 20px + 6px);
+  padding: 10px;
+  gap: 1px;
+}
+
+.calendar__day {
+  align-content: center;
+  box-sizing: border-box;
+  width: 100px;
+  height: 100px;
+  background-color: #eee;
+  border: 1px solid black;
+  position: relative;
+  font-family: Arial, sans-serif;
+  font-size: 30px;
+  transition:
+    background-color 0.5s,
+    transform 0.5s;
+}
+
+.calendar__day--start-day-mon :first-child {
+  margin-left: 0;
+}
+
+.calendar__day--start-day-tue :first-child {
+  margin-left: 101px;
+}
+
+.calendar__day--start-day-wed :first-child {
+  margin-left: 202px;
+}
+
+.calendar__day--start-day-thu :first-child {
+  margin-left: 303px;
+}
+
+.calendar__day--start-day-fri :first-child {
+  margin-left: 404px;
+}
+
+.calendar__day--start-day-sat :first-child {
+  margin-left: 505px;
+}
+
+.calendar__day--start-day-sun :first-child {
+  margin-left: 606px;
+}
```

Comment:
The `margin-left` values for different start days are hardcoded. Consider using a more dynamic approach, possibly using CSS variables or calculations, to make this more maintainable.